File: EJER_ARREGLOS.java
import java.util.ArrayList;
import java.util.Random;
import java.util.Scanner;

public class EJER_ARREGLOS {
    static String[] meses = {"Enero", "Febrero", "Marzo", "Abril", "Mayo", "Junio",
                             "Julio", "Agosto", "Septiembre", "Octubre", "Noviembre", "Diciembre"};
    static ArrayList<String> departamentos = new ArrayList<>();
    static int[][] matriz;
    static Random rand = new Random();
    static Scanner sc = new Scanner(System.in);

    public static void main(String[] args) {
       
        departamentos.add("Ropa");
        departamentos.add("Deportes");
        departamentos.add("Juguetería");

        
        matriz = new int[meses.length][departamentos.size()];
        for (int i = 0; i < meses.length; i++) {
            for (int j = 0; j < departamentos.size(); j++) {
                matriz[i][j] = rand.nextInt(4001) + 1000; 
            }
        }

        
        while (true) {
            System.out.println("\n=== MENÚ DE VENTAS ===");
            System.out.println("1. Mostrar tabla de ventas");
            System.out.println("2. Insertar/Actualizar venta");
            System.out.println("3. Buscar una venta");
            System.out.println("4. Eliminar una venta");
            System.out.println("5. Agregar nuevo departamento");
            System.out.println("6. Salir");
            System.out.print("Elige una opción: ");
            int opcion = sc.nextInt();
            sc.nextLine(); 

            switch (opcion) {
                case 1:
                    mostrarTabla();
                    break;
                case 2:
                    insertarVenta();
                    break;
                case 3:
                    buscarVenta();
                    break;
                case 4:
                    eliminarVenta();
                    break;
                case 5:
                    agregarDepartamento();
                    break;
                case 6:
                    System.out.println("Saliendo del sistema...");
                    return;
                default:
                    System.out.println("Opción no válida. Intenta de nuevo.");
            }
        }
    }

    
    public static void insertarVenta() {
        System.out.print("Número de mes (1-12): ");
        int mes = sc.nextInt() - 1;

        for (int i = 0; i < departamentos.size(); i++) {
            System.out.println((i + 1) + ". " + departamentos.get(i));
        }
        System.out.print("Elige departamento: ");
        int depto = sc.nextInt() - 1;

        System.out.print("Ingresa la venta: ");
        int valor = sc.nextInt();

        matriz[mes][depto] = valor;
        System.out.println("Venta registrada.");
    }

    public static void buscarVenta() {
        System.out.print("Ingresa el valor de la venta a buscar: ");
        int valor = sc.nextInt();

        for (int i = 0; i < meses.length; i++) {
            for (int j = 0; j < departamentos.size(); j++) {
                if (matriz[i][j] == valor) {
                    System.out.println("Venta " + valor + " encontrada en " + meses[i] + " - " + departamentos.get(j));
                    return;
                }
            }
        }
        System.out.println("Venta no encontrada.");
    }

    public static void eliminarVenta() {
        System.out.print("Número de mes (1-12): ");
        int mes = sc.nextInt() - 1;

        for (int i = 0; i < departamentos.size(); i++) {
            System.out.println((i + 1) + ". " + departamentos.get(i));
        }
        System.out.print("Elige departamento: ");
        int depto = sc.nextInt() - 1;

        matriz[mes][depto] = 0;
        System.out.println("Venta eliminada.");
    }

    public static void agregarDepartamento() {
        sc.nextLine(); 
        System.out.print("Nombre del nuevo departamento: ");
        String nombre = sc.nextLine();
        departamentos.add(nombre);

        
        int[][] nuevaMatriz = new int[meses.length][departamentos.size()];
        for (int i = 0; i < meses.length; i++) {
            for (int j = 0; j < departamentos.size() - 1; j++) {
                nuevaMatriz[i][j] = matriz[i][j];
            }
            nuevaMatriz[i][departamentos.size() - 1] = 0;
        }
        matriz = nuevaMatriz;

        System.out.println("Departamento '" + nombre + "' agregado.");
    }

    public static void mostrarTabla() {
        System.out.printf("%-12s", "Mes");
        for (String d : departamentos) {
            System.out.printf("%-12s", d);
        }
        System.out.println();

        
        for (int k = 0; k < 12 * (departamentos.size() + 1); k++) {
            System.out.print("-");
        }
        System.out.println();

        for (int i = 0; i < meses.length; i++) {
            System.out.printf("%-12s", meses[i]);
            for (int j = 0; j < departamentos.size(); j++) {
                System.out.printf("%-12d", matriz[i][j]);
            }
            System.out.println();
        }
    }
}

File: EJER-ARREGLOS.py
import random

class SistemaVentas:
    def __init__(self):
        self.meses = ["Enero", "Febrero", "Marzo", "Abril", "Mayo", "Junio",
                      "Julio", "Agosto", "Septiembre", "Octubre", "Noviembre", "Diciembre"]
        self.departamentos = ["Ropa", "Deportes", "Juguetería"]
        self.matriz = [[random.randint(1000, 5000) for _ in self.departamentos] for _ in self.meses]

    
    def insertar_venta(self, mes, depto, valor):
        self.matriz[mes][depto] = valor
        print(f"Venta de {valor} registrada en {self.meses[mes]} - {self.departamentos[depto]}.")

    def buscar_venta(self, valor):
        for i in range(len(self.matriz)):
            for j in range(len(self.matriz[i])):
                if self.matriz[i][j] == valor:
                    return f"Venta {valor} encontrada en {self.meses[i]} - {self.departamentos[j]}"
        return "Venta no encontrada."

    def eliminar_venta(self, mes, depto):
        self.matriz[mes][depto] = 0
        print(f"Venta eliminada en {self.meses[mes]} - {self.departamentos[depto]}.")

    def agregar_departamento(self, nombre_depto):
        self.departamentos.append(nombre_depto)
        for fila in self.matriz:
            fila.append(0)
        print(f"Departamento '{nombre_depto}' agregado.")

    def mostrar_tabla(self):
        encabezado = f"{'Mes':<12}" + "".join([f"{d:<12}" for d in self.departamentos])
        print(encabezado)
        print("-" * len(encabezado))
        for i in range(len(self.meses)):
            fila = f"{self.meses[i]:<12}" + "".join([f"{self.matriz[i][j]:<12}" for j in range(len(self.departamentos))])
            print(fila)

    def ejecutar_menu(self):
        while True:
            print("\n=== MENÚ DE VENTAS ===")
            print("1. Mostrar tabla de ventas")
            print("2. Insertar/Actualizar venta")
            print("3. Buscar una venta")
            print("4. Eliminar una venta")
            print("5. Agregar nuevo departamento")
            print("6. Salir")

            opcion = input("Elige una opción: ")

            if opcion == "1":
                self.mostrar_tabla()

            elif opcion == "2":
                mes = int(input("Ingresa número de mes (1-12): ")) - 1
                for i, d in enumerate(self.departamentos):
                    print(f"{i+1}. {d}")
                depto = int(input("Elige departamento: ")) - 1
                valor = int(input("Ingresa la venta: "))
                self.insertar_venta(mes, depto, valor)

            elif opcion == "3":
                valor = int(input("Ingresa el valor de la venta a buscar: "))
                print(self.buscar_venta(valor))

            elif opcion == "4":
                mes = int(input("Ingresa número de mes (1-12): ")) - 1
                for i, d in enumerate(self.departamentos):
                    print(f"{i+1}. {d}")
                depto = int(input("Elige departamento: ")) - 1
                self.eliminar_venta(mes, depto)

            elif opcion == "5":
                nombre = input("Ingrese el nombre del nuevo departamento: ")
                self.agregar_departamento(nombre)

            elif opcion == "6":
                print("Saliendo del sistema...")
                break

            else:
                print("Opción no válida. Intenta de nuevo.")


sistema = SistemaVentas()
sistema.ejecutar_menu()
